pull request 7 спринта

ebatuev opened the pull request:
1 коммит 7 спринта

Commit:
1 коммит 7 спринта

pomegranate282922 commented on src/main/java/org/example/CourierClient.java:
```diff
@@ -0,0 +1,65 @@
+                        .body(courier)
+                        .when()
+                        // отправляем POST-запрос с помощью метода post, недостающую часть URL (ручку) передаём в него в качестве параметра
+                        .post("/api/v1/courier");
```
Урлы лучше выносить как константы

ebatuev replied:
Добавлен package constants класс PathAPI - вынес урны как константы 😊

pomegranate282922 commented on src/main/java/org/example/OrderClient.java:
```diff
@@ -0,0 +1,37 @@
+
+    public OrderClient() {
+
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
Base uri во всем проекте один, поэтому нужно его присваивать один раз.
Нужно исправить

ebatuev replied:
Добавлен package actions класс BaseAPI - вынес Base url, так как во всем проекте он один, поэтому нужно его присваивать один раз

pomegranate282922 commented on src/main/java/org/example/Orders.java:
```diff
@@ -0,0 +1,123 @@
+package org.example;
+
+public class Orders {
```
Почему по множественном числе?

ebatuev replied:
Согласен, заказ у нас один формируется, назвал некорректно. 
Переименовал Orders на Order)

pomegranate282922 commented on src/main/java/org/example/RandomCourier.java:
```diff
@@ -0,0 +1,21 @@
+public class RandomCourier extends Courier {
+
+    // Код для генерации произвольных\рандомных значений для логина, пароля и имени курьера
+    private static final String RANDOM_LOGIN = RandomStringUtils.random(10, true, false);
```
Отлично, что используется рандомизация данных

pomegranate282922 commented on src/test/java/org/example/CreateCourierTest.java:
```diff
@@ -0,0 +1,118 @@
+
+    CourierClient courierClient = new CourierClient();
+
+    //Courier courier = new RandomCourier();
```
Закомментированный код нужно удалить

ebatuev replied:
Закомментированный код - удалил 👌

pomegranate282922 commented on src/test/java/org/example/CreateOrdersTest.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    @Parameterized.Parameters // добавили аннотацию
+    public static Object[][] getTestDataForColor() {
```
Отлично, что используется параметризация

ebatuev replied:
В задании была такая задача) в README есть)
Пришлось повозиться немного 😁

pomegranate282922 commented on src/test/java/org/example/GetOrderTest.java:
```diff
@@ -0,0 +1,23 @@
+
+        orderClient.getOrder()
+                .then()
+                .body("orders", notNullValue()); // тело ответа содержит orders; матчер notNullValue() проверяет, что аргумент метода assertThat — не null-значение
```
Лучше проверять непустоту, а не null

ebatuev replied:
Ох, на сколько я помню проверить, пустая ли строка у нас был метод isEmpty() , не соображу что-то, как применить его тут...
Или что-то другое подразумевается?)

ebatuev replied:
еще попробовал вот так реализовать, тоже проходит 

.body("orders", hasSize(greaterThan(0))); // тело ответа содержит orders; матчер hasSize() проверяет, что аргумент — содержит больше чем 0 элементов

pomegranate282922 replied:
Да, это и имела в виду. Реализация может бать разная.

pomegranate282922 commented on src/main/java/org/example/RandomCourier.java:
```diff
@@ -0,0 +1,21 @@
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+public class RandomCourier extends Courier {
```
В итоге этот класс не используется. Можно его удалить.

ebatuev replied:
да, класс RandomCourier в итоге не используется, пометил это в README внизу в ps.
Удалил)

pomegranate282922 reviewed (commented):
Работа отличная! Все тесты написаны, вызовы ручек вынесены в отдельные классы, есть параметризация , данные рандомизированы. Все созданные курьеры удаляются. 
 Всего пара замечаний набралась.

Commit:
2 коммит 7 спринта

1. Добавлен package constants класс PathAPI - вынес урны как константы
2. Добавлен package actions класс BaseAPI - вынес Base url, так как во всем проекте он один, поэтому нужно его присваивать один раз
3. Согласен, заказ у нас один формируется, назвал некорректно. Переименовал Orders на Order
4. да, класс RandomCourier в итоге не используется, пометил это в README внизу в ps.Удалил класс RandomCourier

pomegranate282922 commented on src/main/java/actions/BaseAPI.java:
```diff
@@ -0,0 +1,12 @@
+import io.restassured.specification.RequestSpecification;
+
+public class BaseAPI { // класс Base url во всем проекте один, поэтому присвоили один раз
+    public RequestSpecification RequestSpecification() { // метод RequestSpecification() добавляемм в каждый шаг CourierClient и OrderClient
```
Методы именуем начиная со строчной буквы. Всегда.

ebatuev replied:
ой ей, в ночи писал)) поправлю у себя)

pomegranate282922 replied:
Ага)

pomegranate282922 commented on src/main/java/actions/BaseAPI.java:
```diff
@@ -0,0 +1,12 @@
+public class BaseAPI { // класс Base url во всем проекте один, поэтому присвоили один раз
+    public RequestSpecification RequestSpecification() { // метод RequestSpecification() добавляемм в каждый шаг CourierClient и OrderClient
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru/";
+        return RestAssured.given();
```
Обычно в сам req spec добавляют base uri. А ещё туда можно добавить content type, чтоб ещё уменьшить дублирование кода.

ebatuev replied:
возьму на заметку и поправлю) Спасибо)

pomegranate282922 reviewed (approved):
Нейминг неожиданно захромал. Но будем считать это случайностью.
Критичных замечаний нет.

Commit:
2 коммит 7 спринта

1. Добавлен package constants класс PathAPI - вынес урны как константы
2. Добавлен package actions класс BaseAPI - вынес Base url, так как во всем проекте он один, поэтому нужно его присваивать один раз
3. Согласен, заказ у нас один формируется, назвал некорректно. Переименовал Orders на Order
4. да, класс RandomCourier в итоге не используется, пометил это в README внизу в ps.Удалил класс RandomCourier
